src/Mod/Part/Gui/SectionCutting.cpp

```diff
@@ -137,21 +137,56 @@ SectionCut::SectionCut(QWidget* parent)
     Base::BoundBox3d BoundCutBox;
     if (doc->getObject(BoxXName) || doc->getObject(BoxYName) || doc->getObject(BoxZName)) {
         // automatic coloring must be disabled
-        ui->AutoCutfaceColor->setChecked(false);
+        ui->autoCutfaceColorCB->setChecked(false);
+        ui->autoBFColorCB->setChecked(false);
         if (doc->getObject(CompoundName)) {
+            // get the object with the right name
             auto compoundObject = doc->getObject(CompoundName);
-            Part::Compound* pcCompound = dynamic_cast<Part::Compound*>(compoundObject);
-            if (!pcCompound) {
-                Base::Console().Error(
-                    "SectionCut error: compound is incorrectly named, cannot proceed\n");
-                return;
+            // to later store the childs
+            std::vector<App::DocumentObject*> compoundChilds;
+
+            // check if this is a BooleanFragments or a Part::Compound
+            // Part::Compound is the case when there was only one object
+            Part::Compound* pcCompoundPart = dynamic_cast<Part::Compound*>(compoundObject);
+            if (!pcCompoundPart) {
+                // for more security check for validity accessing its ViewProvider
+                auto pcCompoundBF = Gui::Application::Instance->getViewProvider(compoundObject);
+                if (!pcCompoundBF) {
+                    Base::Console().Error(

```

Comment:
There are many places in this class where you write an error and return from the function. A better way is to throw an exception and handle it in the calling instance. Especially for the overdue refactoring this will make things much easier.

Reply:
> A better way is to throw an exception and handle it in the calling instance

OK. I lack here knowledge - one one hand how it is done and on the other hand the technical insights about the advantage.
Maybe you have a pointer/link for me to read in?

Reply:
Example:
Assuming you have a big function f() and after refactoring you extracted a further functions g(). Inside g() you want to stop the further execution for some reason. With the current approach the code will have this structure:
```
void f()
{
...
   if (g() < 0)
       return;
...
}

int g()
{
...
    if (...) {
        PrintError();
        return -1;
    }
...
}
```
Imagine after a full refactoring you could split some functions into many small and cascaded functions. Now if you want to stop in the innermost function you have to transport the error code to  the calling instance, do this for the next calling instance and so on.
Here error handling is too much bundled with the actual business logic and thus it becomes hardly to maintain. 

By using exceptions you can concentrate on the business logic and handle the exceptions in the outermost function.
```
void f()
{
   try {
      g();
   }
   catch (const exception& e) {
      PrintError(e.what());
   }
}

void g()
{
...
    if (...) {
        throw exception("bla bla bla");
    }
...
}
```
For more details see:
https://docs.oracle.com/javase/tutorial/essential/exceptions/advantages.html
https://www.geeksforgeeks.org/c-exception-handling-question-1/?ref=rp

Reply:
> Example: Assuming you have a big function f() and after refactoring you extracted a further functions g(). Inside g() you want to stop the further execution for some reason...

Many thanks! This is an excellent description and I understood this.
So I will apply this when refactoring.
Due to the pending funded FEM project, I can however, probably first do this in April.